UI: Handle Empty File Uploads in FileUploader

Samk13 opened the pull request:
:heart: Thank you for your contribution!

### Description

* Closes https://github.com/inveniosoftware/invenio-app-rdm/issues/2804
* Depends on https://github.com/inveniosoftware/invenio-app-rdm/pull/2815
* Display a warning for empty files, indicating they won't be included.
* Show a list of removed file names in the warning.
* This feature is controlled by `records-resources-allow-empty-files` config value.
* Continue uploading other files while showing the warning message.

![upload-empty-files](https://github.com/user-attachments/assets/58005efb-0664-477e-9b68-9b8736cf5618)


### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [ ] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [ ] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [ ] I've added relevant test cases.
- [ ] I've added relevant documentation.
- [ ] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/howtos/i18n/).
- [ ] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [ ] I've NOT included third-party code (copy/pasted source code or new dependencies).
    * If you have added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Frontend**

- [ ] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines.
- [ ] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines.
- [ ] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines.


**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

ntarocco commented on invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/FileUploader/FileUploader.js:
```diff
@@ -130,7 +135,24 @@ export const FileUploaderComponent = ({
+              <Message
+                warning
+                icon="warning circle"
+                header="Could not upload files."
+                content={i18next.t("Empty files were skipped as they aren’t allowed.")}
```
I would add translations for the header and remove the part `as they aren’t allowed`, as I think it is kind of normal?
```suggestion
                header=i18next.t("Could not upload all files.")
                content={i18next.t("Empty files were skipped.")}
```

ntarocco commented on invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/FileUploader/FileUploader.js:
```diff
@@ -130,7 +135,24 @@ export const FileUploaderComponent = ({
+        }
+
+        // Proceed with uploading the non-empty files or all files if empty files are allowed
+        if (allowEmptyFiles || !_isEmpty(nonEmptyFiles)) {
```
```suggestion
        if (allowEmptyFiles || hasNonEmptyFiles) {
```

ntarocco reviewed (commented):
Thanks, LGTM! Just a few comments to make it a bit more readable!

Samk13 commented on invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/FileUploader/FileUploader.js:
```diff
@@ -79,6 +81,9 @@ export const FileUploaderComponent = ({
       );
       const maxFileStorageReached = filesSize + acceptedFilesSize > quota.maxStorage;
 
+      const emptyFiles = acceptedFiles.filter((file) => file.size === 0);
+      const nonEmptyFiles = acceptedFiles.filter((file) => file.size > 0);
```
You've removed the arrays emptyFiles and duplicateFiles and replaced them with boolean flags like hasEmptyFiles and hasDuplicatedFiles. But, the arrays emptyFiles and duplicateFiles are necessary because they're used to display the file names in the warning messages. Without these arrays, the app cannot list the specific file names that are empty or duplicated, this is why I added them, just let me know if there's a better way to do it.

Samk13 commented:
@ntarocco  Since you wanted to improve readability, I've added a separate commit where I refactored the component. Now, the uploader continues uploading the rest of the files while showing warnings for duplicate files or empty files (was not possible before). I've also added other missing translations and small fixes.
So you can review this commit separately.
![Screenshot from 2024-09-14 21-57-04](https://github.com/user-attachments/assets/ccfcb5d1-3358-49b4-9445-764e98fd42bf)

ntarocco reviewed (approved):
Fantastic improvement and very nice and readable implementation 👏🏼
